repos/system_upgrade/el7toel8/actors/satellite_upgrader/actor.py

```diff
@@ -19,6 +19,15 @@ def process(self):
         if not facts or not facts.has_foreman:
             return
 
+        if facts.postgresql.local_postgresql:
+            api.current_actor().show_message('Re-indexing the database. This can take a while.')
+            try:
+                run(['sed', '-i', '/data_directory/d', '/var/lib/pgsql/data/postgresql.conf'])
+                run(['systemctl', 'start', 'postgresql'])
+                run(['runuser', '-u', 'postgres', '--', 'reindexdb', '-a'])
+            except (OSError, CalledProcessError) as e:
```

Comment:
From my understanding you might get CalledProcessError during run execution, not OSError

Reply:
Yeah, I think I had it for the installer run because in some intermediate version of the first actor I managed to end up in a situation where the installer binary would not be there and *that* yieleded a `OSError`, can drop because that should really not happen here

Reply:
@fernflower yes, the `run` function can returns both, `OSError` and `CalledProcessError`. The second one happens e.g. when the process exits with non-zero exit code. The first one in case the command cannot be called (e.g. due to missing executable file). So it's usually better to catch both, even when for the core utilities it's expected they are always present. In this case I consider it ok  to catch just CalledProcessError, however, having them both is additional seatbelt.